### PR TITLE
Disable file watcher in VcsManagerTest suite

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

We can't really control the timing of file watcher events, which sometimes leads to failures of VCS tests. The PR disables the file watcher in the `VcsManagerTest` suite to make tests more stable.

Changelog:
- add: a `Watcher` and `WatcherFactory` interfaces
- add: a `NoopWatcher` test watcher
- update: disable the file watcher in the `VcsManagerTest` suite

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by hubertp — APPROVED** [review]
Thanks!

**Review by Akirathan — APPROVED** [review]

File: lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/Watcher.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package org.enso.filewatcher
+
+import java.nio.file.Path
+
+trait Watcher {
```
Comment (JaroslavTulach) [inline lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/Watcher.scala:5]: This PR is a nice opportunity to separate API (e.g. `Watcher`) from the implementation and have them in different projects.

File: lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/WatcherAdapter.scala [hunk 2]
```diff
@@ -12,99 +12,68 @@ import java.nio.file.Path
   */
 final class WatcherAdapter(
   root: Path,
-  eventCallback: WatcherAdapter.WatcherEvent => Unit,
-  errorCallback: WatcherAdapter.WatcherError => Unit
-) extends DirectoryChangeListener {
-
-  import WatcherAdapter._
```
Comment (JaroslavTulach) [inline lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/WatcherAdapter.scala:19]: The test shouldn't even see this class and definitely not ` io.methvin.watcher` classes on the classpath!

File: lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/WatcherFactory.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package org.enso.filewatcher
+
+import java.nio.file.Path
+
+trait WatcherFactory {
```
Comment (JaroslavTulach) [inline lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/WatcherFactory.scala:5]: I'd make the `WatcherFactory` an inner class of `Watcher` if I was designing the API in Java.

**Review by JaroslavTulach — APPROVED** [review]
My wish is to cleanup the dependencies of our project. I believe there should be API (e.g. `Watcher`) visible to the rest of the code base, but the implementation should be in a different project.

My other wish: new files (`Watcher`, `Watcher.Factory`) would better be written in Java.

**Pull request merged**